origin-dapp-creator-server/src/middleware.js

```diff
@@ -11,8 +11,19 @@ export async function validateSubdomain(req, res, next) {
   const { address, config } = req.body
 
   if (config.subdomain) {
+    // Check subdomain doesn't contain invalid characters
+
+    // eslint-disable-next-line no-useless-escape
+    const subdomainRe = /[^a-zA-Z0-9\-]/
```

Comment:
Corner case: looks like subdomain can now contain non-ascii characters:
https://domainname.shop/faq?id=8&session=106ee5e67d523298

But let's not worry about it for now.